starknet_py/tests/e2e/client/client_test.py

```diff
@@ -101,27 +102,33 @@ async def test_get_block_by_hash(
     block_with_deploy_root,
     contract_address,
 ):
-    for client in clients:
-        block = await client.get_block(block_hash=block_with_deploy_hash)
-
-        assert block == StarknetBlock(
-            block_number=block_with_deploy_number,
-            block_hash=block_with_deploy_hash,
-            parent_block_hash=0x0,
-            root=block_with_deploy_root,
-            status=BlockStatus.ACCEPTED_ON_L2,
-            timestamp=2137,
-            transactions=[
-                DeployTransaction(
-                    contract_address=contract_address,
-                    constructor_calldata=[],
-                    hash=deploy_transaction_hash,
-                    signature=[],
-                    max_fee=0,
-                    # class_hash=class_hash,
-                )
-            ],
-        )
+    gateway_client = clients[0]
+    fullnode_client = clients[1]
```

Comment:
There are two fixtures: gateway_client and rpc_client, so you could use them instead of "clients"